### PR TITLE
Small fix for query string encoding

### PR DESCRIPTION
**Pull request by albogdano**
Hey thanks for this handy module!
I've recently noticed that it wasn't escaping some characters properly i.e. `*, !, (, ), '` so I added the method `uriEscape()` to do that.

### CODE REVIEW

**Comment by albogdano** [comment]
Sorry, there's was a typo in my commit - `this.uriEscape = function(str)` should be `function uriEscape(str)`.

File: test/fast.js [hunk 1]
```diff
@@ -157,6 +157,46 @@ describe('aws4', function() {
     })
   })
 
+  describe('#uriEscape()', function() {
+    var uriEscape = function (str) {
```
Comment (mhart) [inline test/fast.js]: I'm not sure I really understand – you're redeclaring the function here and then testing it? This won't actually test the function declared in `aws4.js`

**Comment by mhart** [comment]
Fixed in v1.0.2 – thanks for the report!

File: test/fast.js (hunk 1 above)
Comment (albogdano) [inline test/fast.js]: Yeah, I know, that was stupid but I was in a hurry to commit this. It's the same function, copy-pasted. Sorry about that!

**Pull request closed without merging**